### PR TITLE
fix propagation issue in java-opentracing-server

### PR DESCRIPTION
**Pull request by malafeev**

### CODE REVIEW

File: java/microdonuts/src/main/java/com/otsample/api/ApiContextHandler.java [hunk 1]
```diff
@@ -1,153 +1,147 @@
...
+public class ApiContextHandler extends ServletContextHandler {
+  Properties config;
+  KitchenConsumer kitchenConsumer;
 
-import io.opentracing.Scope;
-import io.opentracing.util.GlobalTracer;
+  public ApiContextHandler() {
+
+    TracingFilter tracingFilter = new TracingFilter(GlobalTracer.get());
+    addFilter(new FilterHolder(tracingFilter), "/*", EnumSet.allOf(DispatcherType.class));
```
Comment (malafeev) [inline java/microdonuts/src/main/java/com/otsample/api/ApiContextHandler.java:34]: adding this filter fixed issue.

Comment (kayousterhout) [reply java/microdonuts/src/main/java/com/otsample/api/ApiContextHandler.java]: can you describe a little more about what this filter is and why it's needed?

Comment (malafeev) [reply java/microdonuts/src/main/java/com/otsample/api/ApiContextHandler.java]: this filter extract span context from request headers and creates a new span using extracted context as parent.
created span is active until request is finished. 
without this filter client span and server spans are not connected.

**Comment by codeboten** [comment]
Based on the output from the integration test I see [here](https://github.com/lightstep/opentelemetry-examples/pull/26/checks?check_run_id=1229496467#step:6:147) and my own testing, I still can't seem to see the requests being propagated to the `java-opentracing-server`, any thoughts?

**Comment by malafeev** [comment]
@codeboten in the integration tests output I see errors from python like:
```
ModuleNotFoundError: No module named 'opentelemetry.sdk.trace.propagation.b3_format.B3Format'; 'opentelemetry.sdk.trace.propagation.b3_format' is not a package
```

```
span_id = span.get_context().span_id
AttributeError: '_Span' object has no attribute 'get_context'
```

I used own java client to test propagation.

**Comment by malafeev** [comment]
```
AssertionError: Services not found: ['java-opentracing-server']
```
does it mean that `java-opentracing-server` is not started?

**Comment by malafeev** [comment]
@codeboten how I can reproduce the issue without running full integration suite? From my java client to server context is propagated.

**Comment by codeboten** [comment]
I run the following to bring up a demo client and the java server, I configured my access token in my .env file as per the readme instructions and i see the 200 response in my request to the `java-opentracing-server`, but the trace ends on the client side. I can see the trace on the server side as well, just not connected to the client. This works just fine for other python client -> java servers (java-specialagent-server, java-otlp-server), not sure what's different.

```
docker-compose up demo-client-otlp java-opentracing-server
```

**Comment by codeboten** [comment]
> ```
> AssertionError: Services not found: ['java-opentracing-server']
> ```
> 
> does it mean that `java-opentracing-server` is not started?

This means that the trace did not contain a report from a service named `java-opentracing-server` in the trace that was created in the integration test

**Comment by malafeev** [comment]
looks like `demo-client-otlp` doesn't send b3 headers to `java-opentracing-server`. I intercepted incoming request from client and I didn't find in request headers any b3 header.

**Comment by codeboten** [comment]
> looks like `demo-client-otlp` doesn't send b3 headers to `java-opentracing-server`. I intercepted incoming request from client and I didn't find in request headers any b3 header.

@malafeev found a bug in the demo-client-otlp that was causing it to propagate via tracecontext instead of b3. i fixed this and pushed the fix to this branch. i can see the b3 headers being sent to the `java-opentracing-server`. still not seeing the trace propagated from the client to the server though. here's the tracing headers. any thoughts?

```
x-b3-traceid: 9c1bbe67a9152dad1ced02a7a32907d3
x-b3-spanid: 058cf03ddba935fb
x-b3-parentspanid: 99e93aafcbeaf8e1
x-b3-sampled: 1
```

**Comment by codeboten** [comment]
@malafeev any thoughts?

**Comment by carlosalberto** [comment]
Btw, we were injecting *single encoding* B3 headers, and we will switch to the (expected) multi-encoding ones (in the upcoming 0.11.0 Launcher Release next week). Not sure this is related, but wanted to mention this anyway ;)

**Comment by malafeev** [comment]
@codeboten I didn't have time yet to check, will do soon.

**Comment by malafeev** [comment]
@codeboten I found the issue: java expects headers with some letters in upper case:
```
  static final String TRACE_ID_HEADER = "X-B3-TraceId";
  static final String SPAN_ID_HEADER = "X-B3-SpanId";
```

and then:
```
 String traceId = getter.get(carrier, TRACE_ID_HEADER);
```

@carlosalberto what can we do with it? 
`x-b3-traceid` is not the same as `X-B3-TraceId`.

**Comment by carlosalberto** [comment]
Cross-writing this comment about this part in the Specification: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/context/api-propagators.md#get

> The Get function is responsible for handling case sensitivity.

Specially important for `B3` as its header keys are expected to be case-insensitive.

**Comment by malafeev** [comment]
@codeboten I believe propagation issue is fixed, please verify.

**Pull request merged**